### PR TITLE
Add new cases of network: update-device

### PR DESCRIPTION
**Pull request by chloerh**
- VIRT-298315 - [update-device] Live update unsupported live update attributes or elements - alter
- VIRT-298316 - [update-device] Live update unsupported live update attributes or elements - add optional setting
- VIRT-298317 - [update-device] Live update unsupported live update attributes or elements - delete the optional elements/attributes

### CODE REVIEW

File: provider/virtual_network/network_base.py [hunk 1]
```diff
@@ -258,3 +258,18 @@ def set_static_ip(iface, ip, netmask, session):
     """
     session.cmd('systemctl stop NetworkManager'),
     session.cmd(f'ifconfig {iface} {ip}/{netmask}')
+
+
+def get_iface_xml_inst(vmxml, comment, index=0):
+    """
+    Get iface xml instance with given vm and index
+
+    :param vm_name: name of vm
```
Comment (dzhengfy) [inline provider/virtual_network/network_base.py:267]: Change to vmxml

File: provider/virtual_network/network_base.py [hunk 2]
```diff
@@ -258,3 +258,18 @@ def set_static_ip(iface, ip, netmask, session):
...
+    :param comment: comment to log
+    :param index: index of interface on vm, defaults to 0
+    :return: xml instance of interface
+    """
+    # vmxml = vm_xml.VMXML.new_from_dumpxml(vm_name)
```
Comment (dzhengfy) [inline provider/virtual_network/network_base.py:272]: remove this line

File: provider/virtual_network/network_base.py [hunk 3]
```diff
@@ -258,3 +258,18 @@ def set_static_ip(iface, ip, netmask, session):
     """
     session.cmd('systemctl stop NetworkManager'),
     session.cmd(f'ifconfig {iface} {ip}/{netmask}')
+
+
+def get_iface_xml_inst(vmxml, comment, index=0):
```
Comment (dzhengfy) [inline provider/virtual_network/network_base.py:263]: This function is too simple. I am wondering if we really need it.

**Review by dzhengfy — COMMENTED** [review]
Others LGTM

File: provider/virtual_network/network_base.py (hunk 1 above)
Comment (chloerh) [reply provider/virtual_network/network_base.py]: Updated.

File: provider/virtual_network/network_base.py (hunk 2 above)
Comment (chloerh) [reply provider/virtual_network/network_base.py]: Updated

**Review by Yingshun — APPROVED** [review]
LGTM

**Review by dzhengfy — APPROVED** [review]

**Pull request merged**